### ISSUE TITLE
"curly": [true, "ignore-same-line"] gives false positives

### ISSUE THREAD

**Issue by aleksey-bykov** [palantir/tslint#3003]
### Bug Report

- __TSLint version__: 5.0.0-dev.0
- __TypeScript version__: 2.4.0-dev.20170607
- __Running TSLint via__: (pick one) CLI: `node ./node_modules/tslint/bin/tslint --project ./scripts/src/tsconfig.json --config ./tslint-standard.json --format msbuild`

#### TypeScript code being linted

```ts
// file: timeline.tsx
if (!this.isReady()) return notReady();
//...
if (this.shouldIgnoreNextScrolled) return void (this.shouldIgnoreNextScrolled = false);
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "semicolon": true,
        "triple-equals": [ true, "allow-null-check" ],
        "switch-default": true,
        "no-switch-case-fall-through": false,
        "max-line-length": [ true, 300 ],
        "no-unused-expression": false,
        "no-use-before-declare": false,
        "quotemark": [ true, "single", "jsx-double" ],
        "curly": [true, "ignore-same-line"],
        "indent": [true, "spaces"],
        "eofline": true,
        "no-inferrable-types": true,
        "no-debugger": true,
        "no-trailing-whitespace": true,
        "one-line": [ true, "check-open-brace" ],
        "align": [ true, "parameters", "statements" ]
    }
}
```

#### Actual behavior
```
scripts/src/parts/timeline.tsx(33,9): error curly: if statements must be braced
scripts/src/parts/timeline.tsx(43,9): error curly: if statements must be braced
```
#### Expected behavior
no linting issues

**Comment by adidahiya** [comment]
I'm not sure that rule option made it into 5.0.0-dev.0, but it is in 5.0.0: https://github.com/palantir/tslint/blob/master/CHANGELOG.md#v500.

I recommend switching to the regular `tslint@latest` release channel; `tslint@next` is deprecated.

**Comment by JoshuaKGoldberg** [comment]
@aleksey-bykov I'm not able to reproduce this error with TSLint 5.11 based an expanded version of your code snippet.

```typescript
// file: timeline.tsx
const notReady = () => true;

class Foo {
    public test() {
        if (!this.isReady()) return notReady();
        //...
        if (this.shouldIgnoreNextScrolled) return void (this.shouldIgnoreNextScrolled = false);
    }

    private isReady(): boolean { return true; }

    private shouldIgnoreNextScrolled = true;
}
```

Closing as aged away. If this is still showing up for you, please post a full file that we can validate against.
